gems/aws-sdk-core/lib/aws-sdk-core/plugins/telemetry.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Aws
+  module Plugins
+    # @api private
+    class Telemetry < Seahorse::Client::Plugin
+      option(
+        :telemetry_provider,
+        default: Aws::Telemetry::NoOpTelemetryProvider,
+        doc_type: Aws::Telemetry::TelemetryProviderBase,
+        rbs_type: 'untyped',
+        docstring: <<~DOCS) do |_cfg|
+Allows you to provide a telemetry provider, which is used
+to emit telemetry data. By default, will use `NoOpTelemetryProvider`
+which will not record or emit any telemetry data.
+        DOCS
+        resolve_provider
```

Comment:
Since this is just a default rather than a resolution (search ENV/shared_config), I'd lean towards getting rid of the `resolve_provider` method here and just returning the NoOp impl.